### PR TITLE
Cluster classes for Azure, AWS and vSphere, first cut

### PR DESCRIPTION
**Pull request by vuil**
### What this PR does / why we need it

```
    - First pass of ClusterClass or Azure, AWS and vSphere and authored ClusterClass patches
      with the same config variable names used earlier in the legacy cluster creation workflow
    - ClusterClass support for "vSphere Windows" is under development
    - New variable "CLUSTER_CLASS" added to specify the ClusterClass to use

    clusterclass-based overlays can be more independently updated

    ClusterClass organization in the providers directory -

    pkg/v1/providers/infrastructure-{aws/azure/vsphere}/{version}/cconly 
       - This directory contains the ClusterClass definitions and the patches
    pkg/v1/providers/infrastructure-{aws/azure/vsphere}/{version}/yttcc 
       - This directory holds the cluster template and the reqired clusterclass
    pkg/v1/providers/infrastructure-{aws/azure/vsphere}/yttcc 
       - This directory also contains clusterclass specfic overlays
    pkg/v1/providers/yttcc 
       - This directory contains the clusterclass specific overlays common across all infrastructures
```

```
    - Add patches for node-labels to all providers (#2161)

    This change updates the kubeletExtraArgs to set node-labels based on
    tkrData, updates imageTag data and updates VM image data for AWS/Azure

    - Add 'required' field to TKR_DATA patch (#2312)

    - Update nodeLabels patch to properly append to map (#2328)

    Makes sure to add the node-labels as an extra item in the
    kubeletExtraArgs map instead of replacing the map.

    - Update cc vars to organize/simplify data values (#2179)

    This change introduces new variable structures to simplify the variables
    that are applied to clusterclass based clusters.

    - Apply overlays based on the 'features.global.package-based-lcm-beta' flag

    and remove unnecessary overlays from yttcc directory

    Co-authored-by: Nick Tenczar <ntenczar@vmware.com>
    Co-authored-by: PremKumar Kalle <pkalle@vmware.com>
    Co-authored-by: Sai Bandi <sai@vmware.com>
    Co-authored-by: Anuj Chaudhari <anujc@vmware.com>
    Co-authored-by: Chandra Pamuluri <cpamuluri@vmware.com>
```

### Which issue(s) this PR fixes

Fixes 1340
Fixes 1341
Fixes 1342

### Describe testing done for PR

Verified that make clustergen producing non-clusterclass configurations are not affected.
The diffs included in the PR comment should also demonstrate the same.
Generation of the clusterclass-based output will be in a followup PR.

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by vuil** [comment]
https://storage.googleapis.com/tkg-clustergen/2510/20220527063613/clustergen.diff.txt
The diffs are harmless, and is a result of these changes in the new output
1. normalizing the order resources by kind and name for comparison
2. having a more accurate variable to infra association, cause a few more entries in the data values resource associated with the cluster

**Comment by alfredthenarwhal** [comment]
> @vuil:  /test install-vc7
Commit: 1a6cd29

Tests passed! Build no: 2081

File: pkg/v1/providers/infrastructure-azure/v1.0.1/cconly/overlay.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#@ load("@ytt:overlay", "overlay")
```
Comment (saji-pivotal) [inline pkg/v1/providers/infrastructure-azure/v1.0.1/cconly/overlay.yaml:1]: @vuil We've merged Azure v1.2.1 to main. We probably should rebase this branch off main and merge. Or merge all the Azure related changes in a separate PR. Thoughts?

**Review by saji-pivotal — CHANGES_REQUESTED** [review]

File: pkg/v1/providers/infrastructure-azure/v1.0.1/cconly/overlay.yaml (hunk 1 above)
Comment (vuil) [reply pkg/v1/providers/infrastructure-azure/v1.0.1/cconly/overlay.yaml]: thanks, rebased and updated to 1.2.1

**Review by saji-pivotal — APPROVED** [review]

**Review by imikushin — APPROVED** [review]
LGTM!

**Pull request merged**